### PR TITLE
Reduce container image size by ditching caches

### PR DESCRIPTION
**Pull request by AlanCoding**
Building before this change

```
ansible/ansible-runner                   devel               881f38e17e7c        56 minutes ago      528MB
```

```
$ docker history 881f38e17e7c
IMAGE               CREATED             CREATED BY                                      SIZE                COMMENT
881f38e17e7c        57 minutes ago      /bin/sh -c #(nop)  CMD ["ansible-runner" "ru…   0B                  
3c6b06d9bb52        57 minutes ago      /bin/sh -c #(nop)  ENTRYPOINT ["entrypoint"]    0B                  
fba81e671cc3        57 minutes ago      /bin/sh -c #(nop) WORKDIR /runner               0B                  
ab56279cbcd3        57 minutes ago      /bin/sh -c #(nop)  ENV HOME=/home/runner        0B                  
db13f22575ed        57 minutes ago      /bin/sh -c #(nop)  ENV RUNNER_BASE_COMMAND=a…   0B                  
b30f8df65707        57 minutes ago      /bin/sh -c #(nop)  ENV LC_ALL=en_US.UTF-8       0B                  
c83842fedd20        57 minutes ago      /bin/sh -c #(nop)  ENV LANGUAGE=en_US:en        0B                  
7880c18d750a        57 minutes ago      /bin/sh -c #(nop)  ENV LANG=en_US.UTF-8         0B                  
05dcd0c9354e        57 minutes ago      /bin/sh -c #(nop)  VOLUME [/runner]             0B                  
e42ff11217c8        57 minutes ago      /bin/sh -c for dir in       /home/runner    …   746B                
90c5a5ad8181        About an hour ago   /bin/sh -c chmod +x /bin/entrypoint             733B                
0b55013ae1e5        About an hour ago   /bin/sh -c #(nop) ADD file:64e34d13b7f0fbb08…   733B                
49b093b1efc7        About an hour ago   /bin/sh -c useradd runner && usermod -aG roo…   297kB               
6db02bf21315        About an hour ago   /bin/sh -c pip3 install bindep https://githu…   42.8MB              
d20947dd64a1        About an hour ago   /bin/sh -c dnf install -y gcc python3-devel     152MB               
e0a21073fcca        About an hour ago   /bin/sh -c dnf install -y epel-release &&   …   118MB               
21087ede6b2e        About an hour ago   /bin/sh -c #(nop) ADD dir:a98bacad1385159f1c…   10B                 
9403fa5adfc1        About an hour ago   /bin/sh -c #(nop) ADD dir:521eaafe2051cad1b2…   157B                
776d412c143b        About an hour ago   /bin/sh -c #(nop) ADD dir:e37610a24753db4db6…   3.45kB              
7618887d33d0        About an hour ago   /bin/sh -c #(nop) ADD 11c0b0c43c7c51af4e3d83…   24.1kB              
0d120b6ccaa8        11 days ago         /bin/sh -c #(nop)  CMD ["/bin/bash"]            0B                  
<missing>           11 days ago         /bin/sh -c #(nop)  LABEL org.label-schema.sc…   0B                  
<missing>           11 days ago         /bin/sh -c #(nop) ADD file:538afc0c5c964ce0d…   215MB
```

----

applied this optimization attempt

```
ansible/ansible-runner                   devel               c6e8d3b4689b        5 seconds ago       461MB
```

```
$ docker history c6e8d3b4689b
IMAGE               CREATED             CREATED BY                                      SIZE                COMMENT
c6e8d3b4689b        34 seconds ago      /bin/sh -c #(nop)  CMD ["ansible-runner" "ru…   0B                  
c1ed4be1f13f        34 seconds ago      /bin/sh -c #(nop)  ENTRYPOINT ["entrypoint"]    0B                  
80f0f4113b4d        34 seconds ago      /bin/sh -c #(nop) WORKDIR /runner               0B                  
c681d221a776        35 seconds ago      /bin/sh -c #(nop)  ENV HOME=/home/runner        0B                  
20ee250f5de4        35 seconds ago      /bin/sh -c #(nop)  ENV RUNNER_BASE_COMMAND=a…   0B                  
f970d386f250        35 seconds ago      /bin/sh -c #(nop)  ENV LC_ALL=en_US.UTF-8       0B                  
a2e95b3be429        35 seconds ago      /bin/sh -c #(nop)  ENV LANGUAGE=en_US:en        0B                  
b039d5534340        35 seconds ago      /bin/sh -c #(nop)  ENV LANG=en_US.UTF-8         0B                  
6142043eddd7        35 seconds ago      /bin/sh -c #(nop)  VOLUME [/runner]             0B                  
c8ffaebbeac2        36 seconds ago      /bin/sh -c for dir in       /home/runner    …   4.83kB              
de36d96ef556        36 seconds ago      /bin/sh -c chmod +x /bin/entrypoint             733B                
a1fcbc0a4ede        37 seconds ago      /bin/sh -c #(nop) ADD file:64e34d13b7f0fbb08…   733B                
b6c47b652073        37 seconds ago      /bin/sh -c useradd runner && usermod -aG roo…   297kB               
32201af6d9ff        38 seconds ago      /bin/sh -c pip3 install --no-cache-dir binde…   31.2MB              
b56b6ebc3951        59 seconds ago      /bin/sh -c dnf install -y epel-release &&   …   214MB               
21087ede6b2e        About an hour ago   /bin/sh -c #(nop) ADD dir:a98bacad1385159f1c…   10B                 
9403fa5adfc1        About an hour ago   /bin/sh -c #(nop) ADD dir:521eaafe2051cad1b2…   157B                
776d412c143b        About an hour ago   /bin/sh -c #(nop) ADD dir:e37610a24753db4db6…   3.45kB              
7618887d33d0        About an hour ago   /bin/sh -c #(nop) ADD 11c0b0c43c7c51af4e3d83…   24.1kB              
0d120b6ccaa8        11 days ago         /bin/sh -c #(nop)  CMD ["/bin/bash"]            0B                  
<missing>           11 days ago         /bin/sh -c #(nop)  LABEL org.label-schema.sc…   0B                  
<missing>           11 days ago         /bin/sh -c #(nop) ADD file:538afc0c5c964ce0d…   215MB 
```

Requesting a review from @wenottingham because I heard he's really into cutting down image size.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -13,10 +13,12 @@ RUN dnf install -y epel-release && \
     dnf install -y python3-pip sudo rsync openssh-clients sshpass glibc-langpack-en git && \
     alternatives --set python /usr/bin/python3 && \
     chmod +x /bin/tini && \
-    rm -rf /var/cache/dnf
+    rm -rf /var/cache/dnf && \
+    dnf install -y gcc python3-devel && \
+    dnf clean all && \
+    rm -rf /var/cache/yum
```
Comment (AlanCoding) [inline Dockerfile]: wait, this is already in there a few lines up, and needs to be dnf, not yum

**Comment by wenottingham** [comment]
I like it!

**Comment by matburt** [comment]
I'm good with this.

**Review by shanemcd — APPROVED** [review]

**Comment by shanemcd** [comment]
Ooops. Looks like this has a conflict now.

File: Dockerfile [hunk 2]
```diff
@@ -13,10 +13,11 @@ RUN dnf install -y epel-release && \
     dnf install -y python3-pip sudo rsync openssh-clients sshpass glibc-langpack-en git && \
     alternatives --set python /usr/bin/python3 && \
     chmod +x /bin/tini && \
+    dnf install -y gcc python3-devel && \
+    dnf clean all && \
```
Comment (shanemcd) [inline Dockerfile:16]: @wenottingham Do you know if this removes anything other than stuff under `/var/cache/dnf` (removed entirely in the next line)?

Comment (wenottingham) [reply Dockerfile:16]: Maybe, but not sure.

**Comment by AlanCoding** [comment]
While I would like to merge this (maybe after auditing the size diff again), I would put #504 in line ahead of it, because that's a top priority.

**Comment by AlanCoding** [comment]
this was somewhat mistaken on a few points, and the whole thing will get absorbed into some new work

**Pull request closed without merging**